src/connection.rs

```diff
@@ -389,29 +389,37 @@ impl rustls_connection {
         }
     }
 
-    /// Retrieves the cipher suite agreed with the peer.
-    /// This returns NULL until the ciphersuite is agreed.
-    /// The returned pointer lives as long as the program.
+    /// Retrieves the cipher suite identifier agreed with the peer.
+    /// This returns 0 until the ciphersuite is agreed.
     /// <https://docs.rs/rustls/latest/rustls/enum.Connection.html#method.negotiated_cipher_suite>
     #[no_mangle]
     pub extern "C" fn rustls_connection_get_negotiated_ciphersuite(
         conn: *const rustls_connection,
-    ) -> *const rustls_supported_ciphersuite {
+    ) -> u16 {
         ffi_panic_boundary! {
-            let conn = try_ref_from_ptr!(conn);
-            let negotiated = match conn.negotiated_cipher_suite() {
+            match try_ref_from_ptr!(conn).negotiated_cipher_suite() {
+                Some(cs) => u16::from(cs.suite()),
+                None => 0,
```

Comment:
💭 0x0000 is `TLS_NULL_WITH_NULL_NULL` which is a good value to use for this. Another option would be 0xffff (in the reserved range) but that is less nice from C, as it's truthy.

Reply:
Replaced to use `TLS_NULL_WITH_NULL_NULL` (_which sounds a bit like a terrible dinner order. "Yeah, can I get NULL with a side of NULL, extra NULL and no mayo?"_)